---
title: "GDSCN Book: SARS with RStudio on AnVIL"
date: "`r format(Sys.time(), '%B %d, %Y')`"
site: bookdown::bookdown_site
documentclass: book
bibliography: book.bib
biblio-style: apalike
link-citations: yes
description: "This book provides resources for instructors to engage students in a cloud-based RStudio activity on AnVIL, focused on the evolutionary history of SARS-CoV-2 variants."
favicon: assets/GDSCN_style/gdscn_favicon.ico
output:
    bookdown::word_document2:
      reference_docx: assets/gdscn-template.docx
      toc: true
---

# Overview {-}

This book provides resources for instructors to engage students in a cloud-based RStudio activity on AnVIL, focused on the evolutionary relationships among the SARS-CoV-2 variants.

There is a growing need for undergraduate students to learn cutting-edge concepts in genomics data science, including performing analysis on the cloud instead of a personal computer. This lesson aims to introduce basic tree building and interpretation using publicly available genetic samples of SARS-CoV-2. Students will be introduced to the sequencing revolution, variants, the basics of tree building and reading phylogenies, and essentials of cloud computing prior to the lab activity. During the lesson, students will work hands-on with RStudio on the [AnVIL](https://anvilproject.org/) cloud computing resource to check data, build trees, and visualize their results.

## Skills Level {-}

<div class = "notice">
_Genetics_  
**Beginner**: minimal genetics knowledge needed
 
_Programming skills_  
**Beginner**: minimal programming experience needed
</div>

## Learning Objectives {-}

Learning objectives for this activity come from the [Genetics Core Competencies](https://genetics-gsa.org/education/genetics-learning-framework/):

- Generate and interpret trees displaying experimental results
- Use bioinformatics to assess genetics data
- Tap into the interdisciplinary nature of science

## AnVIL Collection {-}

Please check out our full collection of AnVIL resources below!

```{r, echo = FALSE, message = FALSE, warning = FALSE}
source("scripts/anvil_repo_table.R")
# Add below any repositories that should not be added
exclude = c(
  "anvil-intro",
  "AnVIL_Template",
  "AnVIL_Template_Test",
  "GDSCN_bookdown_style"
)
# Create table for AnVIL repos detected through the API
knitr::kable(make_anvil_repo_table(exclude), format = "html")
```

