# References 

All sequences were downloaded from NCBI's GenBank. We used sequence data from the following accession numbers: OX315675.1 (omicron.2), OP093374.1 (omicron.4), OP093373.1 (omicron.5), OK091006.1 (delta), NC_045512.2 (Wuhan_reference), OX003129.1 (beta), OX315743.1 (omicron.1), OW998592.1 (alpha), MW911054.1 (gamma).

1. Benson DA, Cavanaugh M, Clark K, Karsch-Mizrachi I, Lipman DJ, Ostell J, Sayers EW. GenBank. Nucleic Acids Res. 2013 Jan;41(Database issue):D36-42. doi: 10.1093/nar/gks1195. Epub 2012 Nov 27. PMID: 23193287; PMCID: PMC3531190.

1. Cella E, Benedetti F, Fabris S, Borsetti A, Pezzuto A, Ciotti M, Pascarella S, Ceccarelli G, Zella D, Ciccozzi M, Giovanetti M. SARS-CoV-2 Lineages and Sub-Lineages Circulating Worldwide: A Dynamic Overview. Chemotherapy. 2021;66(1-2):3-7. doi: 10.1159/000515340. Epub 2021 Mar 18. PMID: 33735881; PMCID: PMC8089399.

1. Paradis E, Claude J, Strimmer K. APE: Analyses of Phylogenetics and Evolution in R language. Bioinformatics. 2004 Jan 22;20(2):289-90. doi: 10.1093/bioinformatics/btg412. PMID: 14734327.

1. Schliep KP. phangorn: phylogenetic analysis in R. Bioinformatics. 2011 Feb 15;27(4):592-3. doi: 10.1093/bioinformatics/btq706. Epub 2010 Dec 17. PMID: 21169378; PMCID: PMC3035803.

SARS-CoV-2 image credit: Davian Ho for the [Innovative Genomics Institute](https://innovativegenomics.org/free-covid-19-illustrations/)
